service/history/workflow/update/update.go

```diff
@@ -305,7 +299,10 @@ func (u *Update) Admit(
 	}
 
 	if !eventStore.CanAddEvent() {
-		return u.CancelIncomplete(ctx, CancelReasonWorkflowCompleted, eventStore)
+		// There shouldn't be any waiters before update is admitted (this func returns).
+		// Call abortWaiters to seal futures and set state to stateAborted.
+		u.abortWaiters(AbortWaiterReasonWorkflowCompleted)
```

Comment:
Sound like this shouldn't actually be necessary since there's no future to begin with?